src/WebJobs.Script/Extensions/FunctionMetadataExtensions.cs

```diff
@@ -33,6 +33,16 @@ public static bool IsHttpInAndOutFunction(this FunctionMetadata metadata)
             return false;
         }
 
+        public static bool IsHttpInFunction(this FunctionMetadata metadata)
+        {
+            if (metadata.InputBindings.Any(b => string.Equals("httptrigger", b.Type, StringComparison.OrdinalIgnoreCase)))
```

Comment:
**note**: follows same logic as the above method `IsHttpInAndOutFunction`. I can turn this string into a constant for just this file (like "IsDirect" and "IsCodeless" in the vars above), but we proably want this cleanup for dev too, not just this feature branch.

Reply:
Why does the `HttpInAndOut` only accept a single input and a single output binding? Is it trying to only filter for a "pure" http trigger?

Reply:
ah that would make sense -- it's only for "http only" scenarios.